### PR TITLE
feat: adding customerAccount field to invoice response closes #71

### LINKED ISSUE #71: Adding customerAccount field to invoice response
Hi

we generate nearly 500-700 invoices per monnth our customers lose the mails or dont't find it anymore. 
So we would store the value and implement in our customer profile a possibility to navigate to this link.

to be changed/added line in Client.js:114
      customerAccount: httpResponse.headers.szlahu_vevoifiokurl,

can this be added?
thanks

### PR DESCRIPTION
**Pull request by laszloerno**
closes #71

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -111,6 +111,7 @@ export class Client {
       invoiceId: httpResponse.headers.szlahu_szamlaszam,
       netTotal: httpResponse.headers.szlahu_nettovegosszeg,
       grossTotal: httpResponse.headers.szlahu_bruttovegosszeg,
+      customerAccount: httpResponse.headers.szlahu_vevoifiokurl,
```
Comment (ert78gb) [inline lib/Client.js:114]:
```suggestion
      customerAccountUrl: httpResponse.headers.szlahu_vevoifiokurl,
```

I think the `Url` suffix more future proof.

**Review by ert78gb — CHANGES_REQUESTED** [review]
Thx for the PR. Please fix the unit test too.

**Comment by laszloerno** [comment]
hi thanks i hope i fixed it now :)

**Comment by ert78gb** [comment]
test is failed

**Comment by laszloerno** [comment]
sorry this is my first time i add something to somebodys code :)

**Review by ert78gb — APPROVED** [review]

**Review by ert78gb — CHANGES_REQUESTED** [review]
No worries. Take your time.

Please add a test case that validates the new `customerAccountUrl` has the expected proper value

**Comment by ert78gb** [comment]
you could run the test on your development environment with `npm run` command

**Comment by ert78gb** [comment]
And please review your commits/PR and the PR should contains only the modification that are relevant. 
Honestly I don't prefer to reformatting, or do other stylish things because it makes complicated to follow the git history.

**Comment by laszloerno** [comment]
> And please review your commits/PR and the PR should contains only the modification that are relevant. Honestly I don't prefer to reformatting, or do other stylish things because it makes complicated to follow the git history.

yes i saw it and disabled format on save. but too late. :( should i delete this PR and create a new without thoose commits?

**Comment by ert78gb** [comment]
not need. leave them. just add the missing unit test

**Comment by laszloerno** [comment]
i will add like this 
```
it('should have `customerAccountUrl` property', async () => {
        const httpResponse = await client.issueInvoice(invoice)
        expect(httpResponse).to.have.property('customerAccountUrl').that.is.oneOf([undefined, 'string']);
      })
```
as this field is filled only if the account has set customer accounts active

**Comment by ert78gb** [comment]
Sry for I don't show the good answer, but I mentored so many developers and I like if they find the answer.

There are 2 option
- the customer has account and in this case the `szlahu_vevoifiokurl` has been filled with the value.  In this case we need a test that ensure the value of the `customerAccountUrl` is filled with the `szlahu_vevoifiokurl` 

- the customer does not have account in this case the `szlahu_vevoifiokurl` is empty.  Unfortunately the szamlazz.hu documentation is not 100% accurate because it does not contain if the field is empty then the value of `szlahu_vevoifiokurl` is empty string or not provided in the header. I think if the `szlahu_vevoifiokurl` is falsy then the `customerAccountUrl` should be `undefined`

I just now had time to properly check the documentation of szamlazz.hu. The response of `issueInvoice` and `reverseInvoice` are contain the `szlahu_vevoifiokurl`. So please extend both method, write the test and extend the documentation in both cases.

I understand to hard to maintain others code but this is the reality.  Most of the time we read others code, documentations and try to understand.
If you don't wanna spend time on it just let me know and I will finish the PR.

**Comment by laszloerno** [comment]
> Sry for I don't show the good answer, but I mentored so many developers and I like if they find the answer.
> 
> There are 2 option
> 
> * the customer has account and in this case the `szlahu_vevoifiokurl` has been filled with the value.  In this case we need a test that ensure the value of the `customerAccountUrl` is filled with the `szlahu_vevoifiokurl`
> * the customer does not have account in this case the `szlahu_vevoifiokurl` is empty.  Unfortunately the szamlazz.hu documentation is not 100% accurate because it does not contain if the field is empty then the value of `szlahu_vevoifiokurl` is empty string or not provided in the header. I think if the `szlahu_vevoifiokurl` is falsy then the `customerAccountUrl` should be `undefined`
> 
> I just now had time to properly check the documentation of szamlazz.hu. The response of `issueInvoice` and `reverseInvoice` are contain the `szlahu_vevoifiokurl`. So please extend both method, write the test and extend the documentation in both cases.
> 
> I understand to hard to maintain others code but this is the reality. Most of the time we read others code, documentations and try to understand. If you don't wanna spend time on it just let me know and I will finish the PR.

non it is fine, as a "hobby" ex. developer i want to come back into that :) i appriciate your help! i will extend it now also for the reversal (i overseen it because im not using this)

**Comment by laszloerno** [comment]
i enhanced also the reverseInvoice. but for this method there were no test cases. I adjusted a little the docu that the response is written for booth cases createinvoice and reverseInvoice. actually it was only for reverse.

**Comment by ert78gb** [comment]
> but for this method there were no test cases.

please wrote them. I am not the original developer of this package. I just maintain it, but every time when I have to modify it I try to add the missing test.
thx

**Comment by laszloerno** [comment]
sorry for the delay but had to much work, i added a test case for reversal + i enhanced the createInvoice case at one place where the customerAccountUrl is missing and once where is it there.

File: tests/client.spec.js [hunk 2]
```diff
@@ -127,6 +136,13 @@ describe('Client', () => {
 
         expect(parseFloat(httpResponse.grossTotal)).is.a('number')
       })
+
+      it('should have `customerAccountUrl` property', async () => {
+        const httpResponse = await client.issueInvoice(invoice)
+        expect(httpResponse).to.have.property('customerAccountUrl').that.is.satisfies((value) => typeof value === 'string' || value === undefined);
```
Comment (ert78gb) [inline tests/client.spec.js:142]:
```suggestion
        expect(httpResponse.customerAccountUrl).to.be.equals('https://www.szamlazz.hu/szamla/fiok/gd82embu556d2qjagzj3s2ijqeqzds4ckhuf')
```

This test ensure the `customerAccountUrl` has been mapped correctly.

File: tests/client.spec.js [hunk 3]
```diff
@@ -172,9 +189,79 @@ describe('Client', () => {
...
+
+      it('should have `grossTotal` property', async () => {
+        const httpResponse = await client.reverseInvoice(reversalRequest)
+
+        expect(parseFloat(httpResponse.grossTotal)).is.a('number')
+      })
+
+      it('should have `customerAccountUrl` property', async () => {
+        const httpResponse = await client.reverseInvoice(reversalRequest)
+        expect(httpResponse).to.have.property('customerAccountUrl').that.is.satisfies((value) => typeof value === 'string' || value === undefined);
```
Comment (ert78gb) [inline tests/client.spec.js:256]:
```suggestion
        expect(httpResponse.customerAccountUrl).to.be.equals('https://www.szamlazz.hu/szamla/fiok/gd82embu556d2qjagzj3s2ijqeqzds4ckhuf')
```

**Review by ert78gb — CHANGES_REQUESTED** [review]

**Review by ert78gb — APPROVED** [review]

**Comment by ert78gb** [comment]
@laszloerno Thank you for your contribution

**Pull request merged**